src/app/elevation/qgselevationprofilelayertreemodel.cpp

```diff
@@ -112,18 +112,46 @@ QVariant QgsElevationProfileLayerTreeModel::data( const QModelIndex &index, int
             }
 
             case QgsMapLayerType::RasterLayer:
-              if ( QgsLineSymbol *lineSymbol = qgis::down_cast< QgsRasterLayerElevationProperties * >( layer->elevationProperties() )->profileLineSymbol() )
+            {
+              QgsRasterLayerElevationProperties *rlProps = qgis::down_cast< QgsRasterLayerElevationProperties * >( layer->elevationProperties() );
+              switch ( rlProps->profileSymbology() )
               {
-                symbol.reset( lineSymbol->clone() );
+                case Qgis::ProfileSurfaceSymbology::Line:
+                  if ( QgsLineSymbol *lineSymbol = rlProps->profileLineSymbol() )
```

Comment:
would it make sense to have virtual methods `profileLineSymbol`, `profileFillSymbol` in `QgsMapLayerElevationProperties` so the cases could be grouped (to avoid code duplication)?

Reply:
Ok, I've refactored to move a lot of common mesh/raster code to "profile surface" subclasses. There was a lot of duplicate code there!

Reply:
nice.
but apparently you can still remove code duplication here by grouping the 2 cases no?